enhancements/cluster-constraints-properties.md

```diff
@@ -0,0 +1,196 @@
+---
+title: cluster constraint and property
+authors:
+  - "@dinhxuanvu"
+reviewers:
+  - "@kevinrizza"
+  - "@joelandford"
+approvers:
+  - "@kevinrizza"
+creation-date: 2021-10-08
+last-updated: 2020-10-11
+status: provisional
+---
+
+# cluster-constraint-and-property
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The dependency resolution in OLM needs to work with cluster constraints in order to effectively install operators that will work properly in the cluster at runtime. Plus, OLM needs to support cluster properties which reflects the runtime information that the cluster may provide and allows operators to depend on those properties if needed to ensure proper installation and operation.
```

Comment:
At this point, we don't know what a "cluster constraint" is. Please define it up front.